src/modules/notifications/notifications.controller.ts

```diff
@@ -39,16 +41,33 @@ export class NotificationsController {
     const userId = user.id;
     return this.notificationsService.markNotificationAsRead(markNotificationAsRead, notification_id, userId);
   }
-  @Delete('/clear')
-  @ApiOkResponse({ type: MarkAllNotificationAsReadResponse, description: 'Notifications cleared successfully.' })
-  @ApiUnauthorizedResponse({ type: MarkAllNotificationAsReadError, description: 'Unauthorized' })
-  @ApiInternalServerErrorResponse({ type: MarkAllNotificationAsReadError, description: 'Internal Server Error' })
-  @ApiOperation({ summary: 'Marks all notifications a read' })
-  async markAllNotificationsAsRead(@Req() request: Request) {
-    const user = request['user'];
-
-    const userId = user.id;
 
-    return this.notificationsService.markAllNotificationsAsReadForUser(userId);
+  @Get()
+  @ApiResponse({
+    status: 200,
+    description: 'Unread notifications retrieved successfully',
+    type: UnreadNotificationsResponseDto,
+  })
+  @ApiResponse({
+    status: 400,
+    description: 'Invalid request or User not found',
+    type: ErrorDto,
+  })
+  @ApiResponse({
+    status: 500,
+    description: 'Failed to retrieve unread notifications',
+    type: ErrorDto,
+  })
+  async getUnreadNotificationsForUser(@Req() req: { user: UserPayload }, @Query('is_read') is_read: string) {
+    const userId = req.user.id;
+    if (is_read === 'false') {
+      const notifications = await this.notificationsService.getUnreadNotificationsForUser(userId);
+      return {
+        status: 'success',
+        message: 'Unread notifications retrieved successfully',
+        status_code: 200,
+        data: notifications,
+      };
+    }
```

Comment:
This method cannot work.

1. is_read is a boolean and should be checked
2. The method should only run when is_read is "false", what happens if is_read is true?

Reply:
If is_read is true, I can return a response indicating there are no unread notifications to retrieve.